app/src/sharedTest/java/org/oppia/android/app/testing/NavigationDrawerActivityTest.kt

```diff
@@ -272,8 +277,193 @@ class NavigationDrawerActivityTest {
+    ).use {
+      it.openNavigationDrawer()
+      onView(withText(R.string.menu_options)).perform(click())
+      it.openNavigationDrawer()
+      onView(withText(R.string.menu_switch_profile)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_cancel))
+        .inRoot(isDialog())
+        .perform(click())
+      it.openNavigationDrawer()
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.fragment_drawer_nav_view))
+        .check(matches(checkNavigationViewItemStatus(NavigationDrawerItem.OPTIONS)))
+    }
+  }
+
+  // TODO(#2535): Remove notation when Robolectric supports multiple times opening of the NavDrawer
+  // Requires opening the NavDrawer several times which is currently not supported on robolectric
+  @RunOn(TestPlatform.ESPRESSO)
+  @Test
+  fun testNavDrawer_goToHelp_openNavDrawer_clickSwitchProfile_clickCancel() {
+    launch<NavigationDrawerTestActivity>(
+      createNavigationDrawerActivityIntent(internalProfileId)
+    ).use {
+      it.openNavigationDrawer()
+      onView(withText(R.string.menu_help)).perform(click())
+      it.openNavigationDrawer()
+      onView(withText(R.string.menu_switch_profile)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_cancel))
+        .inRoot(isDialog())
+        .perform(click())
+      it.openNavigationDrawer()
+      testCoroutineDispatchers.runCurrent()
+      onView(withId(R.id.fragment_drawer_nav_view))
+        .check(matches(checkNavigationViewItemStatus(NavigationDrawerItem.HELP)))
+    }
+  }
+
+  // TODO(#2535): Remove notation when Robolectric supports multiple times opening of the NavDrawer
+  // Requires opening the NavDrawer several times which is currently not supported on robolectric
+  @RunOn(TestPlatform.ESPRESSO)
+  @Test
+  fun testNavDrawer_goToAdmin_openNavDrawer_clickSwitchProfile_clickCancel() {
+    launch<NavigationDrawerTestActivity>(
+      createNavigationDrawerActivityIntent(internalProfileId)
+    ).use {
+      it.openNavigationDrawer()
+      onView(withText(R.string.administrator_controls)).perform(click())
+      it.openNavigationDrawer()
+      onView(withText(R.string.menu_switch_profile)).perform(click())
+      onView(withText(R.string.home_activity_back_dialog_cancel))
+        .inRoot(isDialog())
+        .perform(click())
+      it.openNavigationDrawer()
+      testCoroutineDispatchers.runCurrent()
+      onView(
+        allOf(
+          withText(R.string.administrator_controls),
+          isDescendantOfA(withId(R.id.administrator_controls_linear_layout))
+        )
+      ).check(matches(ViewMatchers.hasTextColor(R.color.highlightedNavMenuItem)))
+    }
+  }
+
+  // TODO(#2535): Remove notation when Robolectric supports multiple times opening of the NavDrawer
+  // Requires opening the NavDrawer several times which is currently not supported on robolectric
+  @RunOn(TestPlatform.ESPRESSO)
+  @Test
+  fun testNavDrawer_clickSwitchProfile_clickCancel_configChange() {
```

Comment:
testNavDrawer_openNavDrawer_switchProfile_cancel_configChange_homeIsSelected

Reply:
follow the same for others.

Reply:
Done